tests/raftstore_cases/test_conf_change.rs

```diff
@@ -591,6 +591,47 @@ fn test_conf_change_safe<T: Simulator>(cluster: &mut Cluster<T>) {
     pd_client.must_remove_peer(region_id, new_peer(2, 2));
 }
 
+fn test_learner_conf_change<T: Simulator>(cluster: &mut Cluster<T>) {
+    let pd_client = Arc::clone(&cluster.pd_client);
+    pd_client.disable_default_operator();
+    let r1 = cluster.run_conf_change();
+    cluster.must_put(b"k1", b"v1");
+    assert_eq!(cluster.get(b"k1"), Some(b"v1".to_vec()));
+
+    // Add voter (2, 2) to region 1.
+    pd_client.must_add_peer(r1, new_peer(2, 2));
+
+    // Add learner (4, 10) to region 1.
+    let engine_4 = cluster.get_engine(4);
+    pd_client.must_add_peer(r1, new_learner_peer(4, 10));
+    cluster.must_put(b"k2", b"v2");
+    must_get_equal(&engine_4, b"k1", b"v1");
+    must_get_equal(&engine_4, b"k2", b"v2");
+
+    // Remove learner (4, 10) from region 1.
+    pd_client.must_remove_peer(r1, new_learner_peer(4, 10));
+    pd_client.must_add_peer(r1, new_learner_peer(4, 10));
+    must_get_equal(&engine_4, b"k2", b"v2");
+
+    // Promote learner (4 ,10) to voter.
+    pd_client.must_add_peer(r1, new_peer(4, 10));
```

Comment:
How do we ensure it becomes a voter?

Reply:
`must_add_peer` will check it's learner or voter when receiving heartbeats.